### PR TITLE
BUG: series.tolist()[0] get Attribute Error

### LINKED ISSUE #415: BUG: series.tolist()[0] get Attribute Error; Fallback cannot display data
### Describe the bug
After fallback to Pandas, xorbits does not display the real data. 
```
/Users/zhengzhou/desktop/xorbits/python/xorbits/pandas/pandas_adapters/core.py:123: RuntimeWarning: Series.tolist will fallback to Pandas
  warnings.warn(
/Users/zhengzhou/desktop/xorbits/python/xorbits/_mars/deploy/oscar/session.py:1873: UserWarning: No existing session found, creating a new local session now.
  warnings.warn(warning_msg)
2023-04-27 17:42:19,529 xorbits._mars.deploy.oscar.local 17543 WARNING  Web service started at http://0.0.0.0:51588
100%|████████████████████████████████████████████████████████| 100.00/100 [00:00<00:00, 29267.35it/s]
100%|██████████████████████████████████████████████████████████| 100.00/100 [00:00<00:00, 170.63it/s]
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[4], line 1
----> 1 ser.tolist()[0]

File ~/desktop/xorbits/python/xorbits/core/adapter.py:149, in wrap_magic_method.<locals>.wrapped(self, *args, **kwargs)
    145 mars_entity = to_mars(self)
    146 if (mars_entity is None) or (
    147     not hasattr(mars_entity, method_name)
    148 ):  # pragma: no cover
--> 149     raise AttributeError(
    150         f"'{self.data.data_type.name}' object has no attribute '{method_name}'"
    151     )
    152 else:
    153     return wrap_mars_callable(
    154         getattr(mars_entity, method_name),
    155         attach_docstring=False,
    156         is_cls_member=True,
    157     )(*args, **kwargs)

AttributeError: 'object_' object has no attribute '__getitem__'
```
### To Reproduce
To help us to reproduce this bug, please provide information below:
```
import xorbits.pandas as pd
d = {'a': 1, 'b': 2, 'c': 3}
ser = pd.Series(data=d, index=['a', 'b', 'c'])
ser.tolist()[0]
```
1. Your Python version 3.10.9
2. The version of Xorbits you use latest
3. Versions of crucial packages, such as numpy, scipy and pandas 1.5.3
4. Full stack of the error.
5. Minimized code to reproduce the error.

### Expected behavior
Output 1

### PR DESCRIPTION
**Pull request by Zhou1213CN**
## Related issue number

Fixes #415 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/core/adapter.py [hunk 1]
```diff
@@ -143,7 +143,9 @@ def wrap_magic_method(method_name: str) -> Callable[[Any], Any]:
     def wrapped(self: DataRef, *args, **kwargs):
         # trigger on condition execution.
         mars_entity = to_mars(self)
-        if (mars_entity is None) or (
+        if hasattr(self, "to_object") and hasattr(self.to_object(), method_name):
```
Comment (qianduoduo0904) [inline python/xorbits/core/adapter.py:146]: The `to_object` which will trigger execution are called twice, is it possible to optimize it?

File: python/xorbits/core/adapter.py [hunk 2]
```diff
@@ -143,7 +143,11 @@ def wrap_magic_method(method_name: str) -> Callable[[Any], Any]:
     def wrapped(self: DataRef, *args, **kwargs):
         # trigger on condition execution.
         mars_entity = to_mars(self)
-        if (mars_entity is None) or (
+
+        data_object = self.to_object() if hasattr(self, "to_object") else None
```
Comment (aresnow1) [inline python/xorbits/core/adapter.py:147]: Does tt mean all magic methods will trigger execution? Not recommended.

**Review by aresnow1 — APPROVED** [review]
LGTM

File: python/xorbits/pandas/pandas_adapters/core.py [hunk 3]
```diff
@@ -92,10 +92,29 @@ def _get_docstring_src_module(self):
         return pd
 
 
+class PandasFetchDataMethodWrapper(PandasClsMethodWrapper):
+    def get_wrapped(self):
+        wrapped = super().get_wrapped()
+
+        def _wrapped(entity: MarsEntity, *args, **kwargs):
```
Comment (qianduoduo0904) [inline python/xorbits/pandas/pandas_adapters/core.py:99]: Does it need to call `functools.wraps` here?

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Review by aresnow1 — APPROVED** [review]

**Pull request merged**